packages/mui-codemod/src/deprecations/backdrop-props/backdrop-props.js

```diff
@@ -0,0 +1,95 @@
+import findComponentJSX from '../../util/findComponentJSX';
+import assignObject from '../../util/assignObject';
+import appendAttribute from '../../util/appendAttribute';
+
+/**
+ * @param {import('jscodeshift').FileInfo} file
+ * @param {import('jscodeshift').API} api
+ */
+export default function transformer(file, api, options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const printOptions = options.printOptions;
+
+  findComponentJSX(j, { root, componentName: 'Backdrop' }, (elementPath) => {
+    let index = elementPath.node.openingElement.attributes.findIndex(
+      (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'TransitionComponent',
+    );
+    if (index !== -1) {
+      const removed = elementPath.node.openingElement.attributes.splice(index, 1);
+      let hasNode = false;
+      elementPath.node.openingElement.attributes.forEach((attr) => {
+        if (attr.name?.name === 'slots') {
+          hasNode = true;
+          assignObject(j, {
+            target: attr,
+            key: 'transition',
+            expression: removed[0].value.expression,
+          });
+        }
+      });
+      if (!hasNode) {
+        appendAttribute(j, {
+          target: elementPath.node,
+          attributeName: 'slots',
+          expression: j.objectExpression([
+            j.objectProperty(j.identifier('transition'), removed[0].value.expression),
+          ]),
+        });
+      }
+    }
+
+    index = elementPath.node.openingElement.attributes.findIndex(
+      (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'TransitionProps',
+    );
+    if (index !== -1) {
+      const removed = elementPath.node.openingElement.attributes.splice(index, 1);
+      let hasNode = false;
+      elementPath.node.openingElement.attributes.forEach((attr) => {
+        if (attr.name?.name === 'slotProps') {
+          hasNode = true;
+          assignObject(j, {
+            target: attr,
+            key: 'transition',
+            expression: removed[0].value.expression,
+          });
+        }
+      });
+      if (!hasNode) {
+        appendAttribute(j, {
+          target: elementPath.node,
+          attributeName: 'slotProps',
+          expression: j.objectExpression([
+            j.objectProperty(j.identifier('transition'), removed[0].value.expression),
+          ]),
+        });
+      }
+    }
```

Comment:
We don't need this one as the `Backdrop` component does not have a `TransitionProps` prop. We should cover `transitionDuration` prop, though. I think it should look something like:

```suggestion
    index = elementPath.node.openingElement.attributes.findIndex(
      (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'transitionDuration',
    );
    if (index !== -1) {
      const removed = elementPath.node.openingElement.attributes.splice(index, 1);
      const updated = j.objectExpression([
          j.objectProperty(j.identifier('transition'), j.objectExpression([
            j.objectProperty(j.identifier('timeout'), removed[0].value.expression),
          ])),
        ]),
      let hasNode = false;
      elementPath.node.openingElement.attributes.forEach((attr) => {
        if (attr.name?.name === 'slotProps') {
          hasNode = true;
          assignObject(j, {
            target: attr,
            key: 'transition',
            expression: updated
          });
        }
      });
      if (!hasNode) {
        appendAttribute(j, {
          target: elementPath.node,
          attributeName: 'slotProps',
          expression:updated
        });
      }
    }
```